mteb/abstasks/AbsTaskHierarchicalClustering.py

```diff
@@ -0,0 +1,117 @@
+from __future__ import annotations
+
+import logging
+import random
+from typing import Any
+
+import numpy as np
+from datasets import DatasetDict
+from sklearn.cluster import MiniBatchKMeans
+from sklearn.metrics.cluster import v_measure_score
+
+from .AbsTask import AbsTask
+
+logger = logging.getLogger(__name__)
+
+
+def evaluate_hierarchical_clustering(
+    embeddings: np.ndarray,
+    labels: list[list[str]],
+    kmean_batch_size: int = 512,
+    max_depth: int = 5,
+    seed: int = 42,
+) -> list[float]:
+    """Evaluates clustering with v_scores on each level of the cluster hierarchy.
+    On each level (up until the maximum depth) every example is evaluated that has that level.
+    """
+    v_scores = []
+    max_depth = min(max_depth, max(map(len, labels)))
+    # Evaluate on each level til max depth
+    for i_level in range(max_depth):
+        level_labels = []
+        # Assign -1 to gold label if the level is not there
+        for label in labels:
+            if len(label) > i_level:
+                level_labels.append(label[i_level])
+            else:
+                level_labels.append(-1)
+        level_labels = np.array(level_labels)
+        valid_idx = level_labels != -1
+        level_labels = level_labels[valid_idx]
+        level_embeddings = embeddings[valid_idx]
+        clustering_model = MiniBatchKMeans(
```

Comment:
So we essentially just evaluate it each level? I think this is fine, but it might be possible to integrate this into FastClustering (allow for either 1 or multiple labels). This way we only have one implementation

Reply:
I will look into it!